src/carto.js

```diff
@@ -68,12 +75,74 @@ const getAttributeFeature = function (sym, feature, ly) {
   return addFunction(fnBody);
 };
 
+const getAlphaColor = function (color, opacity) {
+
+  if (color && opacity) {
+
+    var fn = `
```

Comment:
I have the feeling we are using the wrong approach here, you are doing:

cartocss -> cartoparser.render(cartocss) -> JS -> patch JS for tangran

and it should be

cartocss -> cartoparser.getRenderer('tangram').render(cartocss)